### DISCUSSION TITLE
Allow adding/pasting multiple actions/repository secrets

### DISCUSSION THREAD

**Discussion by danawoodman** [community/community#58682] [Actions]
### Select Topic Area

Product Feedback

### Body

In line with other providers like Vercel and CloudFlare, It would be very helpful to change the secret creation UI to allow adding multiple secrets in one go vs one at a time. This gets very laborious if you, like I did today, have to add 10+ secrets.

My suggestion would be to implement something like the previously mentioned where you can paste the contents of a `.env` file to automatically fill in multiple secrets. This would probably mean not having a separate page for adding a secret and instead do them inline with the editing interface.

An added side effect will also make editing a secret faster too so you don't have to navigate to a new page just to change a secret.

**Comment by salemaljebaly** [comment]
Totally agree — adding secrets one by one is tedious.

I made a simple tool called [EnvSeeder](https://github.com/salemaljebaly/EnvSeeder) that reads a .env file and pushes all keys to GitHub Actions Secrets in one go via CLI.

Until GitHub supports bulk editing natively, this helps streamline the process.

**Reply by MillerAdulu** [reply]
Used this @salemaljebaly. Massive script. Thank you. 🔥

**Comment by MikaTech-dev** [comment]
might make a chrome extension to automate the process, it's really unnecessarily tedious. I don't want to have to install github cli to add secrets en-masse

**Reply by salemaljebaly** [reply]
We started with the CLI because it works everywhere any OS, CI/CD, direct API calls.
The extension approach has real limits: Chrome only, slower (automates the form one secret at a time instead of API calls), and can't be scripted.
But for a quick visual tool without touching a terminal — the extension wins.

**Comment by salemaljebaly** [comment]
@MikaTech-dev 
Built it — SecretDrop, a Chrome extension that bulk-uploads your .env to GitHub Actions Secrets. No CLI, no API token, drop the file and done.
https://github.com/salemaljebaly/gh-env-extension
